bindings/nodejs/lib/wallet/wallet-method-handler.ts

```diff
@@ -22,11 +22,19 @@ import { SecretManager } from '../secret_manager';
 export class WalletMethodHandler {
     methodHandler: any;
 
+    /**
+     * @param methodHandler The Rust method handler created in `WalletMethodHandler.create()`.
+     */
+    constructor(methodHandler: any) {
+        this.methodHandler = methodHandler;
+    }
+
     /**
      * @param options The wallet options.
      */
-    constructor(options?: WalletOptions) {
-        this.methodHandler = createWallet(JSON.stringify(options));
+    static async create(options: WalletOptions): Promise<WalletMethodHandler> {
+        const methodHandler = await createWallet(JSON.stringify(options));
```

Comment:
Can this not be the one contained in `Wallet` already ? Do we have to wrap it ?

Reply:
Then you can't use `WalletMethodHandler` without creating a wallet first
We could also remove `WalletMethodHandler` completely and only have `Wallet`, but feels a bit cleaner to me to have them separated

Reply:
You could do it, but you miss the function definitions from the JS side https://github.com/iotaledger/iota-sdk/blob/bd7c08f8b9a065312058a2279e161e8193f2a35a/bindings/nodejs/lib/wallet/wallet-method-handler.ts#L24

Reply:
I see, thanks